File: server/src/main/java/com/ny/cfss/notes/edm/individual/Individual.java
package com.ny.cfss.notes.edm.individual;

import com.ny.cfss.notes.edm.AbstractDomainEntity;
import com.ny.cfss.notes.edm.individual.goal.Goal;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "individuals")
@Data
@EqualsAndHashCode(callSuper = true, onlyExplicitlyIncluded = true)
public class Individual extends AbstractDomainEntity {

    @Column(nullable = false)
    private String firstName;

    private String middleName;

    @Column(nullable = false)
    private String lastName;

    @OneToMany(mappedBy = "individual")
    private Set<Goal> goals = new HashSet<>();

}

File: client/src/app/individuals/individual.ts
import {DomainEntity} from "../shared/domain/domain-entity";

export class Individual extends DomainEntity {

  firstName: string;
  middleName: string;
  lastName: string;

  resourceName(): string {
    return "individuals";
  }

  lastNameFirst(): string {
      return `${this.lastName}, ${this.firstName}`;
  }

}
